LinkedList.java

```diff
@@ -46,17 +46,42 @@ public void printLinkedList(LinkedList list) {
         System.out.println("null");
     }
 
+    public void deleteKey(LinkedList list, int key){
+        if(list.head == null){
+            return;
+        }
+        if(list.head.data == key){
+            list.head = list.head.next;
```

Comment:
This is fine. But the deleted node is still pointing to head. Remove that pointer.